caproto/_state.py

```diff
@@ -248,10 +248,11 @@ def _fire_command_triggered_transitions(self, role, command_type):
         try:
             new_state = allowed_transitions[command_type]
         except KeyError:
-            err = get_exception(role, command_type)
-            raise err(
-                "{} cannot handle command type {} when role={} and state={}"
-                .format(self, command_type.__name__, role, self.states[role]))
+            err_cls = get_exception(role, command_type)
+            err = err_cls(f"{self} cannot handle command type "
+                          f"{command_type.__name__} when role={role} and "
+                          f"state={self.states[role]}")
+            raise err from None
```

Comment:
h11 does this, IIRC. Not sure why I originally decided not to imitate that. Makes sense.